gammapy/cube/tests/test_psf_kernel.py

```diff
@@ -0,0 +1,62 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import pytest
+import numpy as np
+from numpy.testing import assert_allclose
+import astropy.units as u
+from astropy.coordinates import Angle
+from ...maps import WcsNDMap, MapAxis, WcsGeom
+from .. import PSFKernel, table_psf_to_kernel_map
+from ...irf import TablePSF, EnergyDependentTablePSF
+
+
+def test_table_psf_to_kernel_map():
+    sigma = 0.5 * u.deg
+    binsz = 0.1 * u.deg
+    geom = WcsGeom.create(binsz=binsz, npix=150)
+
+    rad = Angle(np.linspace(0., 3 * sigma, 100), 'deg')
+    table_psf = TablePSF.from_shape(shape='gauss', width=sigma, rad=rad)
+    kernel = table_psf_to_kernel_map(table_psf, geom, normalize=True)
+
+    # Is normalization OK?
+    assert_allclose(kernel.data.sum(), 1.0, atol=1e-5)
+
+    # maximum at the center of map?
+    ind = np.unravel_index(np.argmax(kernel.data, axis=None), kernel.data.shape)
+    # absolute tolerance at 0.5 because of even number of pixel here
+    assert_allclose(ind, geom.center_pix, atol=0.5)
+
+
+def test_psf_kernel_from_gauss():
+    sigma = 0.5 * u.deg
+    binsz = 0.1 * u.deg
+    geom = WcsGeom.create(binsz=binsz, npix=150, axes=[MapAxis((0, 1, 2))])
+
+    kernel = PSFKernel.from_gauss(geom, sigma)
+
+    # Check that both maps are identical
+    assert_allclose(kernel.to_map().data[0], kernel.to_map().data[1])
+
+    # Is there an odd number of pixels
+    assert_allclose(np.array(kernel.to_map().geom.npix) % 2, 1)
+
+
+def test_psf_kernel_convolve():
+    sigma = 0.5 * u.deg
+    binsz = 0.05 * u.deg
+
+    testmap = WcsNDMap.create(binsz=binsz, width=5 * u.deg)
+    testmap.fill_by_coord(([1], [1]), weights=np.array([2]))
+
+    kernel = PSFKernel.from_gauss(testmap.geom, sigma)
+
+    conv_map = kernel.apply(testmap)
```

Comment:
Does this FFT convolve work properly?
Suggest to do a e.g. (10 x 5) map (different nx / ny) and put a value of 1 or so in the center and at the edges or corners, and to assert on the resulting pixels at the same positions that come out.
Something that really establishes what the PSF apply does, also at the edge of the image.